components/builder-core/src/http_client.rs

```diff
@@ -0,0 +1,98 @@
+use std::collections::HashMap;
+
+use reqwest::{header::HeaderMap,
+              Body,
+              Client,
+              Method,
+              Proxy,
+              Response};
+
+use url::Url;
+
+use crate::error::{Error,
+                   Result};
+
+pub const BLDR_USER_AGENT: &str = "Habitat-Builder";
```

Comment:
Why is this defined here, components/builder-core/src/api_client.rs and components/builder-core/src/rpc.rs? Similarly for other constants.

Along with consolidating the definitions, it may make sense to make them `HeaderValue`s, rather than `&str`, since it appears all the usages are of the form `HeaderValue::from_static(BLDR_USER_AGENT)`. Once [generalized `const fn`s are available](https://github.com/rust-lang/rust/issues/57563),  `HeaderValue::from_static` will probably become one, allowing a declaration like:
```rust
pub const BLDR_USER_AGENT: HeaderValue = HeaderValue::from_static("Habitat-Builder");
```
but until then, we could leverage `lazy_static`:
```rust
lazy_static::lazy_static! {
    pub static ref BLDR_USER_AGENT: HeaderValue= HeaderValue::from_static("Habitat-Builder");
    pub static ref APPLICATION_JSON: HeaderValue= HeaderValue::from_static("application/json");
    // ... any other commonly used `HeaderValue`
}
```
Then the references change from `HeaderValue::from_static(BLDR_USER_AGENT)` to `BLDR_USER_AGENT.clone()`, increasing type safety.

In fact, we could go one step further, and ensure the appropriate pairing of `HeaderName` with `HeaderValue`:
```rust
lazy_static::lazy_static! {
    pub static ref USER_AGENT_BLDR: (HeaderName, HeaderValue) = (USER_AGENT, HeaderValue::from_static("Habitat-Builder"));
    // ...
}
```
Which, together with the suggestion about eliminating the need for `mut` on the `HeaderMap` binding, could make
```rust
    let mut headers = HeaderMap::new();
    headers.insert(USER_AGENT, HeaderValue::from_static(BLDR_USER_AGENT));
    headers.insert(ACCEPT, HeaderValue::from_static(APPLICATION_JSON));
    headers.insert(CONTENT_TYPE, HeaderValue::from_static(APPLICATION_JSON));
```
into:
```rust
    let header_values = vec![USER_AGENT_BLDR.clone(),
                             ACCEPT_APPLICATION_JSON.clone(),
                             CONTENT_TYPE_APPLICATION_JSON.clone()];
    let headers = HeaderMap::from_iter(header_values.into_iter());
```

Reply:
Yep, will fix.